### PR TITLE
Changing GCP Facts modules to Info Modules

### PR DESCRIPTION
**Pull request by rambleraptor**
##### SUMMARY

This ports all of the `gcp_*_facts` modules to `gcp_*_info` modules.

This change:
* Adds deprecation notices to the modules.
* Changes the facts modules to have an underscore in their name
* Changes the facts modules to be a symlink to info modules.
* Creates info modules (that are same as facts modules)
* Changes test + examples to use info modules.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

gcp_*_facts

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by felixfontein** [comment]
Why did you remove `gcp_compute_autoscaler_info.py ` and `gcp_compute_snapshot_info.py` in f8ac5acf898c94c9b3ff33424d6f0922c1006d4f?

Also, you are deleting `_gcp_target_proxy.py` and `_gcp_url_map.py`. Why are you doing that? These should only be removed in Ansible 2.12 (and even then, a stub will remain informing that they have been deleted, AFAIK).

Finally, you can ignore the "More than 50 changed files" complaint by ansibot, I'll merge this manually once it is finished :)

File: lib/ansible/modules/cloud/google/gcp_pubsub_topic.py
Comment (felixfontein) [inline lib/ansible/modules/cloud/google/gcp_pubsub_topic.py]: This new feature requires a changelog entry.

File: lib/ansible/modules/cloud/google/gcp_pubsub_topic.py
Comment (felixfontein) [inline lib/ansible/modules/cloud/google/gcp_pubsub_topic.py]: Please add a note that this return value was added for Ansible 2.9.

File: lib/ansible/modules/cloud/google/gcp_pubsub_topic.py
Comment (felixfontein) [reply lib/ansible/modules/cloud/google/gcp_pubsub_topic.py]: (Alternatively, move this new feature out of this PR.)

File: lib/ansible/modules/cloud/google/gcp_pubsub_topic_info.py
Comment (felixfontein) [inline lib/ansible/modules/cloud/google/gcp_pubsub_topic_info.py]: Also here: this should probably not be part of this PR.

**Comment by rambleraptor** [comment]
whoops, didn't mean to delete those files! Fixed! 

As for `gcp_compute_autoscaler_info` and `gcp_compute_snapshot_info`, those aren't yet merged into Ansible. It didn't seem fully appropriate to add those modules in with this PR.

I've got PRs [here](https://github.com/ansible/ansible/pull/60097) and [here](https://github.com/ansible/ansible/pull/60099) to add those modules in.

**Comment by felixfontein** [comment]
Symlinks which shouldn't be there (since they are for modules added in 2.9):
 * _gcp_appengine_firewall_rule_facts.py
 * _gcp_cloudscheduler_job_facts.py
 * _gcp_cloudtasks_queue_facts.py
 * _gcp_mlengine_model_facts.py
 * _gcp_mlengine_version_facts.py

**Comment by felixfontein** [comment]
You also need a changelog fragment for this PR; see for example https://github.com/ansible/ansible/blob/devel/changelogs/fragments/56822-facts-info-rename.yaml

**Comment by rambleraptor** [comment]
good call! fixes made.

**Review by felixfontein — APPROVED** [review]
I think this is done, except that `gcp_compute_autoscaler_facts` is now missing (#60097).

**Comment by rambleraptor** [comment]
ah, the joys of not force pushing after a rebase! that should fix it.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
If you think this is done too, I'll ask some one from docs team to look over the porting guide entries and then merge it later today.

**Comment by felixfontein** [comment]
(Assuming CI doesn't complain, that is ;) )

File: changelogs/fragments/gcp-facts-info-rename.yaml
Comment (felixfontein) [inline changelogs/fragments/gcp-facts-info-rename.yaml]:
```suggestion
- The ``gcp_bigquery_dataset_facts`` module was renamed to ``gcp_bigquery_dataset``.
```
Sorry, one more. Since the changelog is not part of the docsite, ReST links won't work.

**Review by felixfontein — CHANGES_REQUESTED** [review]

File: changelogs/fragments/gcp-facts-info-rename.yaml
Comment (rambleraptor) [reply changelogs/fragments/gcp-facts-info-rename.yaml]: no problem! made the change on all of the changelog fragments.

**Review by felixfontein — APPROVED** [review]

File: lib/ansible/modules/cloud/google/gcp_appengine_firewall_rule_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_appengine_firewall_rule_info.py]: These auto-generated `name` lines are a bit strange - could you change them to use the pattern "get info on a/an {{ item }}"?

File: lib/ansible/modules/cloud/google/gcp_appengine_firewall_rule_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_appengine_firewall_rule_info.py]: Missed a change here - this module still uses `facts` in the description and doesn't have the "was previously called" line.

File: lib/ansible/modules/cloud/google/gcp_bigquery_dataset_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_bigquery_dataset_info.py]: Please add a period to the end of each line with this pattern (The usage has not changed.).

File: lib/ansible/modules/cloud/google/gcp_bigquery_table_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_bigquery_table_info.py]: We don't need both "previously" and "before Ansible 2.9". How about using the pattern "Before Ansible 2.9, this module was called {{ old_name }}."?

**Review by acozine — COMMENTED** [review]
Wow, this was a huge undertaking. Thanks for making all these changes to conform to our naming conventions! I see a few ways to make the documentation better - notes below.

File: lib/ansible/modules/cloud/google/gcp_appengine_firewall_rule_info.py
Comment (felixfontein) [reply lib/ansible/modules/cloud/google/gcp_appengine_firewall_rule_info.py]: That's because the module was added after stable-2.8 was branched, so it will first be released in Ansible 2.9

File: lib/ansible/modules/cloud/google/gcp_bigquery_table_info.py
Comment (felixfontein) [reply lib/ansible/modules/cloud/google/gcp_bigquery_table_info.py]: Alternatively, there's `This module was called C({{ old_name }}) before Ansible 2.9.`, which I've used in a lot of other renames :)

File: lib/ansible/modules/cloud/google/gcp_bigquery_dataset_info.py
Comment (felixfontein) [inline lib/ansible/modules/cloud/google/gcp_bigquery_dataset_info.py]:
```suggestion
- This module was called C(gcp_bigquery_dataset_facts) before Ansible 2.9. The usage has not changed.
```
Sorry for the confusion; we used `{{ old_name }}` to indicate the old name of the module, assuming that you would replace it while updating the modules. Ansible will simply typeset this as-is and the user will read `{{ old_name }}` and not the real old name of the module.

**Review by felixfontein — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/google/gcp_bigquery_dataset_info.py
Comment (rambleraptor) [reply lib/ansible/modules/cloud/google/gcp_bigquery_dataset_info.py]: Ah, got it! I assumed there was some Jinja magic happening somewhere. Fix made!

File: lib/ansible/modules/cloud/google/gcp_compute_autoscaler_info.py
Comment (felixfontein) [inline lib/ansible/modules/cloud/google/gcp_compute_autoscaler_info.py]:
```suggestion
- name: get info on an autoscaler
```
I guess such details should be left for another PR :)

**Comment by rambleraptor** [comment]
@felixfontein fixes made! ptal

**Review by felixfontein — APPROVED** [review]

File: lib/ansible/modules/cloud/google/gcp_compute_instance.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_instance.py]:
```suggestion
          configured.
```

File: lib/ansible/modules/cloud/google/gcp_compute_instance.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_instance.py]:
```suggestion
        will select the default network interface, which could have SSH already configured.
```

File: lib/ansible/modules/cloud/google/gcp_compute_instance.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_instance.py]:
```suggestion
```

File: lib/ansible/modules/cloud/google/gcp_compute_instance_group_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_instance_group_info.py]:
```suggestion
- name: get info on an instance group
```

File: lib/ansible/modules/cloud/google/gcp_compute_instance_group_manager_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_instance_group_manager_info.py]:
```suggestion
- name: get info on an instance group manager
```

File: lib/ansible/modules/cloud/google/gcp_compute_instance_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_instance_info.py]:
```suggestion
- name: get info on an instance
```

File: lib/ansible/modules/cloud/google/gcp_compute_instance_template_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_instance_template_info.py]:
```suggestion
- name: get info on an instance template
```

File: lib/ansible/modules/cloud/google/gcp_compute_interconnect_attachment_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_interconnect_attachment_info.py]:
```suggestion
- name: get info on an interconnect attachment
```

File: lib/ansible/modules/cloud/google/gcp_compute_ssl_certificate_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_ssl_certificate_info.py]:
```suggestion
- name: get info on an SSL certificate
```

File: lib/ansible/modules/cloud/google/gcp_compute_ssl_policy_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_ssl_policy_info.py]:
```suggestion
- name: get info on an SSL policy
```

File: lib/ansible/modules/cloud/google/gcp_compute_target_http_proxy_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_target_http_proxy_info.py]:
```suggestion
- name: get info on a target HTTP proxy
```

File: lib/ansible/modules/cloud/google/gcp_compute_target_https_proxy_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_target_https_proxy_info.py]:
```suggestion
- name: get info on a target HTTPS proxy
```

File: lib/ansible/modules/cloud/google/gcp_compute_target_ssl_proxy_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_target_ssl_proxy_info.py]:
```suggestion
- name: get info on a target SSL proxy
```

File: lib/ansible/modules/cloud/google/gcp_compute_target_tcp_proxy_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_target_tcp_proxy_info.py]:
```suggestion
- name: get info on a target TCP proxy
```

File: lib/ansible/modules/cloud/google/gcp_compute_target_vpn_gateway_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_target_vpn_gateway_info.py]:
```suggestion
- name: get info on a target VPN gateway
```

File: lib/ansible/modules/cloud/google/gcp_compute_url_map_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_url_map_info.py]:
```suggestion
- name: get info on a URL map
```

File: lib/ansible/modules/cloud/google/gcp_compute_vpn_tunnel_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_vpn_tunnel_info.py]:
```suggestion
- name: get info on a VPN tunnel
```

File: lib/ansible/modules/cloud/google/gcp_redis_instance_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_redis_instance_info.py]:
```suggestion
- name: get info on an instance
```

File: lib/ansible/modules/cloud/google/gcp_spanner_instance_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_spanner_instance_info.py]:
```suggestion
- name: get info on an instance
```

File: lib/ansible/modules/cloud/google/gcp_sql_instance_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_sql_instance_info.py]:
```suggestion
- name: get info on an instance
```

File: lib/ansible/modules/cloud/google/gcp_compute_image_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_image_info.py]:
```suggestion
- name: get info on an image
```

File: lib/ansible/modules/cloud/google/gcp_compute_https_health_check_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_https_health_check_info.py]:
```suggestion
- name: get info on an HTTPS health check
```

File: lib/ansible/modules/cloud/google/gcp_compute_http_health_check_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_http_health_check_info.py]:
```suggestion
- name: get info on an HTTP health check
```

File: lib/ansible/modules/cloud/google/gcp_compute_autoscaler_info.py
Comment (acozine) [reply lib/ansible/modules/cloud/google/gcp_compute_autoscaler_info.py]: oh dear, I'm adding all kinds of comments like this one -  I say let's fix them while we're looking

File: lib/ansible/modules/cloud/google/gcp_compute_address_info.py
Comment (acozine) [inline lib/ansible/modules/cloud/google/gcp_compute_address_info.py]:
```suggestion
- name: get info on an address
```

**Review by acozine — COMMENTED** [review]
I found a lot of nits - mostly a/an and capitalization stuff. I made them all suggestions, so if you just hit accept down the line, it should be ready to merge. Thanks for seeing this one through!

**Comment by rambleraptor** [comment]
@acozine just merged the suggestions. ptal. thanks a lot!

**Review by felixfontein — APPROVED** [review]

**Comment by acozine** [comment]
@rambleraptor thanks! you missed one suggestion, though - the one to gcp_compute_ssl_policy_info.py. Can you accept that last one? 

The bot mysteriously thinks you need a rebase, but I think it's just behind the times . . . I can remove that label before I merge.

**Comment by rambleraptor** [comment]
whoops, thanks for that! suggestion accepted

**Comment by felixfontein** [comment]
@acozine it wants rebase/revision because the PR has more than 50 modified files. But when you merge manually, you can also just ignore the labels...

@rambleraptor @acozine thanks a lot for all your effort in this! I'm happy that this issue is now resolved :-)

**Pull request merged**
